instrumentation/action_pack/lib/opentelemetry/instrumentation/action_pack/handlers/action_controller.rb

```diff
@@ -23,9 +23,34 @@ def initialize(config)
           # @param payload [Hash] the payload passed as a method argument
           # @return [Hash] the payload passed as a method argument
           def start(_name, _id, payload)
-            span_name, attributes = to_span_name_and_attributes(payload)
+            request = payload[:request]
+            # It seems that there are cases in Rails functional tests where it bypasses the routing system and the `action_dispatch.route_uri_pattern` header not being set.
+            # Our Test suite executes the routing system so we are unable to recreate this error case.
+            # https://github.com/rails/rails/blob/747f85f200e7bb2c1a31b4e26e5a5655e2dc0cdc/actionpack/lib/action_dispatch/http/request.rb#L160
+            http_route = request.route_uri_pattern&.chomp('(.:format)') if request.respond_to?(:route_uri_pattern)
+
+            attributes = {
+              OpenTelemetry::SemanticConventions::Trace::CODE_NAMESPACE => String(payload[:controller]),
+              OpenTelemetry::SemanticConventions::Trace::CODE_FUNCTION => String(payload[:action])
```

Comment:
The old behaviour would allow this to be overwritten with the exceptions app class and action, so what is currently here keeps that. Do we want to revisit having this also preserve the original controller/action?

I'm a bit split, because it gives a clear bread crumb that an exceptions app handled it, but that's not really obvious...

Reply:
I don't have a strong opinion about how class naming should work. 

If someone configures an exception app they are intentionally trying to perform an internal redirect to another controller. 

We don't have a way a to model what the user intended on doing and what the server actually executed if it is configured to handle requests to the same routes differently.

For me that is where the semconv naming is most helpful because as long as the route is resolved, then we could add additional attributes that describe the controller action or middleware that eventually handled the request.

Reply:
Yeah I think I'll leave it as is. 

>For me that is where the semconv naming is most helpful because as long as the route is resolved, then we could add additional attributes that describe the controller action or middleware that eventually handled the request.

I agree, but there's just such a strong convention of thinking about rails apps in the form of controllers and actions, that we want to continue to support that ease of association. 

If I really wanted to make an argument against my position, the rack and controller span should be separate spans as they are quite different.